pkg/webhook/mutation.go

```diff
@@ -153,6 +160,7 @@ func (h *mutationHandler) Handle(ctx context.Context, req admission.Request) adm
 	}
 
 	resp, err := h.mutateRequest(ctx, req)
```

Comment:
can we skip mutation request if no mutation policy is defined?

Reply:
I sort of asked the same question here https://github.com/open-policy-agent/gatekeeper/pull/1015#discussion_r539469397
As per the comment, it's gonna be handled in a separate pr.

Reply:
If no mutation policy is defined at all (e.g. zero resources declared), this call becomes a no-op as we would be looping over an empty list.

Any skipping or other optimizations that depend on installed mutations should be done within the `Mutate()` function so we can rely on the read lock to guarantee consistency.

Reply:
Ah, this is the handler's mutate request (sorry). There is some serialization overhead, but if a user has no mutations they ever want to apply could they not just disable the mutation webhook entirely? That would yield the biggest latency benefits.

Reply:
I think this depends on what we publish by default, will a default deployment have mutation enabled or will they have to opt in?

Reply:
I think the question of what the default is is out-of-scope for this code, as that depends on the config mechanism (Helm, operator, proprietary thing). We should not couple our code to the decisions made by arbitrary and unknown config mechanisms.

Because the latency of serializing/deserializing is likely much less than the latency of the network round trip (not tested, but likely true as we are already serializing/deserializing the request object to form and respond to the request), attempting to proactively disable it based off of the contents of the controller would be adding complexity/brittleness to the code for negligible benefit.

How users self-identify as interested/uninterested in mutation and whether mutation is opt-in or opt-out is a question for Helm Charts, Operators, and whoever else is providing managed solutions for Gatekeeper.

Reply:
If it turns out that there is some marginal improvement that's worth grabbing, we should find that out empirically, otherwise it's not clear to me the benefit is worth the cost.